File: generic-method.go
package main

import "fmt"

type hello struct {
	times int
}

func (self hello) print(value interface{}) {
	for i := 0; i < self.times; i++ {
		fmt.Println(value)
	}
}

func main() {
	h := hello{2}
	h.print("cool")
	h.print(11)
}

File: generic-method.rs
struct Hello {
    times: i32,
}

impl Hello {
    fn print<T: std::fmt::Display>(&self, value: T) {
        for _ in 0 .. self.times {
            value.fmt();
        }
    }
}

fn main() {
    let h = Hello{ times: 2 };
    h.print("cool");
    h.print(11);
}
